lib/node_modules/@stdlib/blas/base/drotg/test/test.assign.js

```diff
@@ -0,0 +1,242 @@
+	t.end();
+});
+
+tape( 'the function computes a Givens plane rotation', function test( t ) {
+	var expected;
+	var values;
+	var delta;
+	var tol;
+	var out;
+	var i;
+	var j;
+
+	expected = [
+		[ 0.5, 1.0 / 0.6, 0.6, 0.8 ],
+		[ 0.5, 0.6, 0.8, 0.6 ],
+		[ 0.5, -1.0 / 0.6, -0.6, 0.8 ],
+		[ -0.5, -0.6, 0.8, -0.6 ],
+		[ -0.5, 1.0 / 0.6, 0.6, 0.8 ],
+		[ 0.0, 0.0, 1.0, 0.0 ],
+		[ 1.0, 1.0, 0.0, 1.0 ],
+		[ 1.0, 0.0, 1.0, 0.0 ]
+	];
+	values = [
+		[ 0.3, 0.4 ],
+		[ 0.4, 0.3 ],
+		[ -0.3, 0.4 ],
+		[ -0.4, 0.3 ],
+		[ -0.3, -0.4 ],
+		[ 0.0, 0.0 ],
+		[ 0.0, 1.0 ],
+		[ 1.0, 0.0 ]
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		expected[i] = new Float64Array( expected[i] );
+		out = new Float64Array( 4 );
+		drotg.assign( values[i][0], values[i][1], out, 1, 0 );
+		if ( out === expected[i] ) {
+			t.deepEqual( out, expected[i], 'returns expected value' );
+		} else {
+			for ( j = 0; j < out.length; j++ ) {
+				delta = abs( out[j] - expected[i][j] );
+				tol = 1.5 * EPS * abs( expected[i][j] );
+				t.ok( delta <= tol, 'within tolerance. out: '+out[j]+'. expected: '+expected[i][j]+'. delta: '+delta+'. tol: '+tol+'.' );
+			}
+		}
+	}
+	t.end();
+});
+
+tape( 'the function returns an array of NaNs if provided a rotation elimination parameter equal to NaN', function test(t) {
+	var actual;
+	var i;
+
+	actual = drotg( NaN, 1.0, new Float64Array( 4 ), 1, 0 );
+	for ( i = 0; i < actual.length; i++ ) {
+		t.strictEqual( isnan( actual[i] ), true, 'returns expected value' );
+	}
+
+	actual = drotg( 1.0, NaN, new Float64Array( 4 ), 1, 0 );
+	for ( i = 0; i < actual.length; i++ ) {
+		t.strictEqual( isnan( actual[i] ), true, 'returns expected value' );
+	}
+
+	actual = drotg( NaN, NaN, new Float64Array( 4 ), 1, 0 );
+	for ( i = 0; i < actual.length; i++ ) {
+		t.strictEqual( isnan( actual[i] ), true, 'returns expected value' );
+	}
+	t.end();
+});
+
+tape( 'the function supports providing a positive stride', function test(t) {
+	var expected;
+	var actual;
+	var delta;
+	var tol;
+	var out;
+	var i;
+
+	expected = new Float64Array( [ 0.5, 0.0, 1.0 / 0.6, 0.0, 0.6, 0.0, 0.8 ] );
+	out = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
+
+	actual = drotg.assign( 0.3, 0.4, out, 2, 0 );
+	if ( out === expected ) {
+		t.strictEqual( actual, out, 'returns expected value' );
+	} else {
+		for ( i = 0; i < out.length; i++ ) {
+			delta = abs( out[i] - expected[i] );
+			tol = 1.5 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. out: '+out[i]+'. expected: '+expected[i]+'. delta: '+delta+'. tol: '+tol+'.' );
+		}
+	}
+	if ( actual === expected ) {
+		t.deepEqual( actual, expected, 'returns expected value' );
+	} else {
+		for ( i = 0; i < actual.length; i++ ) {
+			delta = abs( actual[i] - expected[i] );
+			tol = 1.5 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. actual: '+actual[i]+'. expected: '+expected[i]+'. delta: '+delta+'. tol: '+tol+'.' );
+		}
+	}
```

Comment:
These tests do not work as expected. The `out === expected` comparison is comparing references, not values. These tests and below need to be refactored.

Reply:
See refactoring above.